scripts/scripts.js

```diff
@@ -377,6 +377,55 @@ export function addFavIcon(href) {
   }
 }
 
+/**
+ * Function to set head meta tags.
+ */
+export function setMetaTag(tagType, propertyKey, propertyValue, url) {
+  const tag = document.querySelector(`${tagType}[${propertyKey}='${propertyValue}']`);
+  if (tag) {
+    if (tagType === 'link') {
+      tag.href = url;
+    } else {
+      tag.content = url;
+    }
+  } else {
+    const meta = document.createElement(tagType);
+    meta.setAttribute(propertyKey, propertyValue);
+    if (tagType === 'link') {
+      meta.href = url;
+    } else {
+      meta.content = url;
+    }
+    document.head.appendChild(meta);
+  }
+}
+
+/**
+ * Function to set following meta tags for tag page
+ *  og:image
+ *  og:image:secure_url
+ *  twitter:image
+ *  og:url
+ *  canonical
+ */
+function setTagPageMetaImages(main) {
+  const pageType = getMetadata('pagetype');
+  if (pageType && pageType.trim().toLowerCase() === 'tagpage') {
+    const images = [...main.querySelectorAll('.fragment-container .cards.block > ul > li img')];
```

Comment:
I think we can generally fetch the first image in the `main` part of the page's markup. The above code relies on a specific fragment being present which is true for the current tag page but if someone were to change its document, it may not be present.

Reply:
Agree, but if someone authors a block which contain any images that would be picked up, let me remove `.fragment-container` and keep card block class there.